blocks/api/categories.js

```diff
@@ -28,3 +38,48 @@ const categories = [
 export function getCategories() {
 	return categories;
 }
+
```

Comment:
It's also possible to achieve a very similar result using this code:

```js
import { applyFilters } from '@wordpress/hooks';

let cachedCategories;
export function getCategories() {
    if ( ! cachedCategories ) {
        const extraCategories = ...applyFilters( 'blocks.getExtraCategories',  {} );
        // we can perform optonal validation here and filter out all items that don't have slug and name 
        cachedCategories = { ...extraCategories, ...categories );
    }        
    return cachedCategories;
}```

Reply:
also, since JavaScript functions are much cheaper than PHP functions we can wrap via composition and keep different logic separate :smile:

```js
import { once } from 'lodash'

const getCategories = once( () => [
	...applyFilters( 'blocks.getExtraCategories', [] ),
	...categories,
] )
```

and without `lodash` it's easy to recreate `once`

```js
const once = f => {
	let isPrimed = false
	let value

	return ( ...args ) => ! isPrimed
		? ( isPrimed = true, value = f( ...args ) )
		: value
}
```

Reply:
@dmsnell thanks for making it even simpler :)